### ISSUE TITLE
test_functional[symlink_module0] fails on any Windows checkout without symlink privilege

### ISSUE THREAD

**Issue by iam-kira** [pylint-dev/pylint#11359]
### Bug description

```python
`tests/functional/s/symlink/_binding/symlink_module.py` and `.../_binding/__init__.py`
are tracked as symlinks (`git ls-files -s` reports mode `120000`). On Windows, git sets
`core.symlinks=false` unless the user is elevated or has Developer Mode enabled, and
materialises those entries as regular text files containing the link target:

    $ cat tests/functional/s/symlink/_binding/symlink_module.py
    ../symlink_module/symlink_module.py

pylint lints that one-line path as Python, emits `syntax-error`, and the functional test
fails. It is the only failure in a full run on Windows.

Why CI doesn't catch it: the Windows job in tests.yaml runs the full suite on
windows-latest and is green on main. Since this test only passes when the symlink is
materialised, those runners evidently do materialise it. The failure appears only on an
ordinary contributor's machine — green in CI, red locally, permanently.

Suggested fix: pylint already has machinery for this — a `.rc` file with
`[testoptions]` / `exclude_platforms=`, enforced at lint_module_test.py:136, already used
by bad_char_carriage_return.rc and super_init_not_called_extensions_py310.rc. Adding
tests/functional/s/symlink/_binding/symlink_module.rc with `exclude_platforms=win32`
makes the case skip cleanly; the sibling symlink_module test still runs and passes
(892 passed, 40 skipped, 0 failed locally).

Known cost: that also skips on CI Windows, where the test currently passes. If you'd
rather keep that coverage, documenting the requirement (Developer Mode, or cloning with
`git -c core.symlinks=true`) is the alternative.

Note for anyone reaching for it: skipping on `not Path(test_file.source).is_symlink()`
does not work — the sibling `symlink_module/symlink_module.py` is a real file on every
platform and never a symlink, so that condition would skip a legitimately passing test
everywhere.

Happy to open a PR with whichever you prefer.
```

### Configuration

```ini

```

### Command used

```shell
pytest tests/
```

### Pylint output

```python
1 failed, 2182 passed, 262 skipped, 5 xfailed
FAILED tests/test_functional.py::test_functional[symlink_module0]

E       AssertionError: Wrong message(s) raised for "symlink_module.py":
E
E       Unexpected in testdata:
E          1: syntax-error (times 1)
```

### Expected behavior

The test is skipped when the working tree could not materialise the symlink, rather than failing.

### Pylint version

```shell
pylint 4.1.0-dev0
astroid 4.2.0b5
Python 3.13.15
```

### OS / Environment

Windows 11 (10.0.26200)
git core.symlinks=false (the default for a non-elevated Windows user)
repo at ef292792

### Additional dependencies

```python

```

**Comment by webdevsamran** [comment]
Hi team, I would like to fix this Windows test failure.

When `core.symlinks=false` on Windows, git materializes tracked symlinks under `tests/functional/s/symlink/_binding/` as 1-line text files, causing `test_functional[symlink_module0]` to fail. I can add a check verifying `os.path.islink()` (or Windows symlink privilege) to cleanly skip this functional test when symlinks are not materialized as links, with full test passes on Windows and Linux.

Could you please assign this to me? Thanks!
